src/viam/sdk/tests/mocks/mock_robot.cpp

```diff
@@ -265,9 +269,14 @@ std::vector<FrameSystemConfig> mock_proto_config_response() {
 }
 
 ::grpc::Status MockRobotService::FrameSystemConfig(
-    ::grpc::ServerContext*,
+    ::grpc::ServerContext* context,
     const ::viam::robot::v1::FrameSystemConfigRequest*,
     ::viam::robot::v1::FrameSystemConfigResponse* response) {
+    auto client_info = context->client_metadata().find("viam_client");
+    if (client_info->second != k_version) {
+        return ::grpc::Status(::grpc::StatusCode::FAILED_PRECONDITION,
+                              "viam_client info not properly set in metadata");
+    }
```

Comment:
These robot tests gave the easiest access to server-side context such that we can confirm that the `viam_client` header is properly being set.